### PR TITLE
feature/add-available-kudos-to-all-employees

### PR DESCRIPTION
**Pull request by rniedziolka**
Hi,
the next feature is ready:
https://www.loom.com/share/86c8755aeab446d08ac4c8db489300d2
Waiting for the feedback!

### CODE REVIEW

File: app/controllers/admin/employees_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      redirect_to admin_employees_url, notice: 'Employee was successfully destroyed.'
+    end
+
+    def edit_kudos_for_all
+      render :update_kudos_for_all, locals: { employees: Employee.all }
+    end
+
+    def update_kudos_for_all
+      if add_kudos_param >= 1 && add_kudos_param <= 20
+        Employee.all.each do |employee|
```
Comment (oskarlakner) [inline app/controllers/admin/employees_controller.rb]: what if the operation would stop in the middle you'd end up with few updated users and few not updated users :). Have a look at the db transactions :)

**Review by oskarlakner — APPROVED** [review]
🙌

File: app/controllers/admin/employees_controller.rb (hunk 1 above)
Comment (rniedziolka) [reply app/controllers/admin/employees_controller.rb]: We're calling the transaction method on the ActiveRecord::Base class and passing it a block. Every database operation that happens inside that block will be sent to the database as a transaction.

If any kind of unhandled error happens inside the block, the transaction will be aborted, and no changes will be made to the DB.

```
ActiveRecord::Base.transaction do
            Employee.find_each do |employee|
              employee.number_of_available_kudos += add_kudos_param
              employee.save!
            end
end
```

**Comment by rniedziolka** [comment]
Hi @oskarlakner,
fixes added: https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/27/commits/b36682253c11ca0cba4bd8316f3bd18a2cde5240 https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/27/commits/59066a823b797ad1f1d1477f6972e09e57a049ef https://github.com/rniedziolka/employee-recognition-platform-1637255604/pull/27/commits/1241ff923148748295db962e33a939c2e8549690 
Could you please take a look?
Thanks!

**Pull request merged**